server/urls.js

```diff
@@ -24,6 +21,7 @@ function isImage(url){
 		request(url, {timeout: RESPONSE_TIMEOUT})
 			.on('response', function(res){
 				res.destroy();
+				reject('No data in response');
```

Comment:
Would a response event always fire after a data event?
If a response event fires before a data event then this may incorrectly reject the promise.